tfx/components/base/base_executor_test.py

```diff
@@ -37,21 +44,130 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
     pass
 
 
+class _BeamTestExecutor(base_executor.BeamExecutor):
+  """Fake BeamExecutor for testing purpose only."""
+
+  def beam_io_signature(self, input_dict: Dict[Text, List[types.Artifact]],
+                        output_dict: Dict[Text, List[types.Artifact]],
+                        exec_properties: Dict[Text, Any]
+                        ) -> List[Dict[Text, Any]]:
+    input_signature = {_EXAMPLES: io.TextIOBase}
+    output_signature = {_SHUFFLED_EXAMPLES: io.TextIOBase}
+    return input_signature, output_signature
+
+  def read_inputs(self, pipeline: beam.Pipeline,
+                  input_dict: Dict[Text, List[types.Artifact]],
+                  output_dict: Dict[Text, List[types.Artifact]],
+                  exec_properties: Dict[Text, Any]
+                  ) -> Dict[Text, beam.pvalue.PCollection]:
+    input_dir = input_dict[_EXAMPLES].uri
+    beam_inputs = ({_EXAMPLES: (pipeline
+                                | beam.io.textio.ReadFromText(input_dir))})
+    return beam_inputs
+
+  def construct_beam_graph(self, pipeline: beam.Pipeline,
+                           beam_inputs: Dict[Text, beam.pvalue.PCollection],
+                           input_dict: Dict[Text, List[types.Artifact]],
+                           output_dict: Dict[Text, List[types.Artifact]],
+                           exec_properties: Dict[Text, Any]
+                           ) -> Dict[Text, beam.pvalue.PCollection]:
+    beam_outputs = {_SHUFFLED_EXAMPLES: (beam_inputs[_EXAMPLES]
+                                         | beam.Map(str.strip)
+                                         | beam.Map(int)
+                                         | beam.CombineGlobally(sum))}
+    return beam_outputs
+
+  def write_outputs(self, pipeline: beam.Pipeline,
+                    beam_outputs: Dict[Text, beam.pvalue.PCollection],
+                    input_dict: Dict[Text, List[types.Artifact]],
+                    output_dict: Dict[Text, List[types.Artifact]],
+                    exec_properties: Dict[Text, Any]) -> None:
+    output_dir = output_dict[_SHUFFLED_EXAMPLES].uri
+    _ = (beam_outputs[_SHUFFLED_EXAMPLES]
+         | beam.io.textio.WriteToText(output_dir))
+
+  def Do(self, input_dict: Dict[Text, List[types.Artifact]],
```

Comment:
@charlesccychen Should Do be defined here or on `FuseableBeamExecutor`?

Reply:
I believe `Do()` should be defined in the `_BeamTestExecutor` class, since the generic implementation of `read_inputs()`, `run_component()`, `write_outputs()` is not possible in the components without extreme refactoring